packages/grid/veilid/server/veilid_streamer.py

```diff
@@ -0,0 +1,441 @@
+# stdlib
+import asyncio
+from collections.abc import Callable
+from collections.abc import Coroutine
+from enum import ReprEnum
+from enum import nonmember
+import hashlib
+import logging
+import math
+from struct import Struct
+from typing import Any
+import uuid
+
+# third party
+import veilid
+
+# relative
+from .constants import MAX_MESSAGE_SIZE
+from .utils import retry
+
+logger = logging.getLogger(__name__)
+logger.setLevel(level=logging.INFO)
+
+# An asynchronous callable type hint that takes bytes as input and returns bytes
+AsyncReceiveStreamCallback = Callable[[bytes], Coroutine[Any, Any, bytes]]
+CallId = bytes
+
+
+class RequestType(bytes, ReprEnum):
```

Comment:
Q: What is the use of ReprEnum?

Reply:
This will create a sort of `BytesEnum` class, similar to the inbuilt `enum.IntEnum` and `enum.StrEnum`. Helps in validation and allows directly accessing members instead of needing to use `enum_member.value` in all places (otherwise we would need to do `RequestType.STREAM_SINGLE.value` and `RequestType.STREAM_CHUNK.value` instead of `RequestType.STREAM_SINGLE` and `RequestType.STREAM_CHUNK`.

To further improve readability, I'll add a separate `BytesEnum` class and inherit the `RequestType` class from that instead.

Reply:
Ahh ok makes sense